source/val/BasicBlock.h

```diff
@@ -123,6 +123,18 @@ class BasicBlock {
   /// Adds @p next BasicBlocks as successors of this BasicBlock
   void RegisterSuccessors(const std::vector<BasicBlock*>& next = {});
 
+  /// Set the successors to this block, without updating other internal state,
+  /// and without updating the other blocks.
+  void SetSuccessorsUnsafe(std::vector<BasicBlock*>&& others) {
+    successors_ = others;
```

Comment:
Yes, indeed, you're right. Since the "others" variable is an lvalue itself.  Good catch!  (rvalue ref stuff is really subtle!)